onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc

```diff
@@ -1115,39 +1115,51 @@ Status MIGraphXExecutionProvider::Compile(const std::vector<FusedNodeAndGraph>&
     }
 
     std::vector<std::string> input_names, output_names;
-    no_input_shape = get_input_output_names(graph_body_viewer, input_names, output_names);
+    no_input_shape = no_input_shape or get_input_output_names(graph_body_viewer, input_names, output_names);
```

Comment:
Too much python lately?
```suggestion
    no_input_shape = no_input_shape || get_input_output_names(graph_body_viewer, input_names, output_names);
```

Reply:
How can you tell? XD

Reply:
Removed